layers/+lang/python/packages.el

```diff
@@ -319,7 +319,10 @@
                    'spacemacs//pyenv-mode-set-local-version)))
       ;; setup shell correctly on environment switch
       (dolist (func '(pyenv-mode-set pyenv-mode-unset))
-        (advice-add func :after 'spacemacs/python-setup-everything))
+        (advice-add func :after
+                    #'(lambda (&optional version)
+                        (spacemacs/python-setup-everything
+                         (when args (pyenv-mode-full-path version))))))
```

Comment:
Yo, the `args` here is unbound. It should probably be `(when version ...` I'm guessing?

Reply:
Yes, that's the root cause. Resolved in https://github.com/syl20bnr/spacemacs/pull/15856. Thanks.